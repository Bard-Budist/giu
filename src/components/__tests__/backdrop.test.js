// @flow

/* eslint-env jest */

import React from 'react';
import renderer from 'react-test-renderer';
import Backdrop from '../backdrop';

describe('Backdrop', () => {
  it('01 renders with default style', () => {
    const tree = renderer.create(
      <Backdrop />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('02 renders with custom style', () => {
    const tree = renderer.create(
      <Backdrop style={{ backgroundColor: 'black' }} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
