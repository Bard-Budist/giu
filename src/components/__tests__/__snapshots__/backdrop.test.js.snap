exports[`Backdrop 01 renders with default style 1`] = `
<div
  className="giu-backdrop"
  onTouchMove={[Function]}
  onWheel={[Function]}
  style={
    Object {
      "backgroundColor": "white",
      "height": "100vh",
      "left": 0,
      "opacity": 0.7,
      "position": "fixed",
      "top": 0,
      "width": "100vw",
    }
  } />
------------HTML PREVIEW---------------
<div
  class="giu-backdrop"
  style="background-color:white;height:100vh;left:0;opacity:0.7;position:fixed;top:0;width:100vw;"
>
</div>
`;

exports[`Backdrop 02 renders with custom style 1`] = `
<div
  className="giu-backdrop"
  onTouchMove={[Function]}
  onWheel={[Function]}
  style={
    Object {
      "backgroundColor": "black",
      "height": "100vh",
      "left": 0,
      "opacity": 0.7,
      "position": "fixed",
      "top": 0,
      "width": "100vw",
    }
  } />
------------HTML PREVIEW---------------
<div
  class="giu-backdrop"
  style="background-color:black;height:100vh;left:0;opacity:0.7;position:fixed;top:0;width:100vw;"
>
</div>
`;
